### ISSUE TITLE
Chrome back navigation replaces all frames

### ISSUE THREAD

**Issue by jodyheavener** [davatron5000/FitVids.js#191]
It's become apparent to me that when in Google Chrome (37), if I navigate away from a page that has multiple FitVids instances, and then use the browser's back button to return to the page, all of the frames become the **same frame** (even though they still contain the original markup).

There is an open Chromium bug for this: https://code.google.com/p/chromium/issues/detail?id=395791

As a temporary fix I have done the following:

    $allVideos.each(function(){
        ...
        // Reload iFrame to fix Chrome bug
        $this.attr('src', $this.attr('src'));
    });

I know it seems silly to have to reset the same source, but it does work. Is there a more permanent fix for this coming down the line, or are we going to hold out until Google fixes the issue with Chrome? It's been there since at least v34 and is still on v37.

**Comment by SteveChab** [comment]
Hey, I found a [really easy fox for this on a Vimeo forum](https://vimeo.com/forums/topic:107023). Just add a unique name attribute to each iframe.


So, if you have two iframes, do this:

`<iframe name="trippy" width="560" height="315" src="https://www.youtube.com/embed/-PefmeS3-dA?rel=0&amp;showinfo=0" frameborder="0" allowfullscreen></iframe>`

`<iframe name="sheep" width="420" height="315" src="https://www.youtube.com/embed/ssAE-CXoNU8?rel=0&amp;showinfo=0" frameborder="0" allowfullscreen></iframe>`

**Comment by davatron5000** [comment]
Does scripting the `name` attribute after DOMContentLoaded work or does it need to be hard coded? 




If that works this is a happy day.

**Comment by SteveChab** [comment]
It seems that the name has to load before DOMContentLoaded. Adding it with JS didn't work for me. Maybe adding a unique name attribute could just be part of the instructions for FitVids?

**Comment by chriscoyier** [comment]
_Chris watches this thread with great interest._

**Comment by davatron5000** [comment]
I can play around with this on Monday. If it works as a solution you get $50 to a charity of your choice!

**Comment by jodyheavener** [comment]
What a great thread this become.

**Comment by SteveChab** [comment]
And just an FYI, I've seen this problem in Safari as well.

**Comment by SteveChab** [comment]
Ever have any luck with this?

**Comment by davatron5000** [comment]
No. Been busy.

**Comment by vdecree** [comment]
Just come across this myself. Getting it in both Firefox and Safari. Firefox seems fine. Any known fixes yet?

**Comment by gomezisdan** [comment]
Was having this issue in chrome only, can confirm that specifying an iFrame name has fixed my problem.

**Comment by monmak2** [comment]
Had this problem in Chrome only... can also confirm that the ONLY solution is to assign a "name" to the iFrame.

For example, this is the normal code : 

```html
<iframe src="https://www.youtube.com/embed/eO26fmb2yoo" width="560" height="315" ></iframe>
```

And this is what you change it to : 

```html
<iframe src="https://www.youtube.com/embed/eO26fmb2yoo" width="560" height="315" name="youtubevideo"></iframe>
```

**Comment by gomezisdan** [comment]
Google should really release this as a note somewhere in their Chrome Docs

**Comment by monmak2** [comment]
Google indeed should release this as a note in docs. I don't normally use iFrames and I tried every method known to man to make this work. Adding 'name' was the only thing that worked.

**Comment by davatron5000** [comment]
Has anyone tried injecting a name? You could modify the following:

https://github.com/davatron5000/FitVids.js/blob/master/jquery.fitvids.js#L72-L75

From `id` to `name` and if it solves the back button issue in Chrome we could roll a fix. But if we can't script a fix, it's moot to add it.

**Comment by chriscoyier** [comment]
uggh i'm the worst here, but i THINK injecting a name doesn't work but having a name from the get-go DOES work.

**Comment by fastlinemedia** [comment]
I just submitted a PR for this and it works for me in the latest version of Chrome... https://github.com/davatron5000/FitVids.js/pull/231

**Comment by spigotdesign** [comment]
Using @fastlinemedia's patch has solved the issue for me.

**Comment by davatron5000** [comment]
Fixed in #231
